Building API documentation should be disabled when Doxygen is not installed.

### Bug description
_A clear and concise description of what the bug is._

```
==> Processing plain cmake package: 'beluga'
==> cmake /mypath/src/beluga/beluga -DCMAKE_INSTALL_PREFIX=/mypath/devel_isolated/beluga -G Unix Makefiles in '/mypath/build_isolated/beluga/devel'
-- Build testing enabled.
-- Build API documentation enabled.
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Configuring done
-- Generating done
-- Build files have been written to: /mypath/build_isolated/beluga/devel
==> make -j4 -l4 in '/mypath/build_isolated/beluga/devel'
Scanning dependencies of target beluga_docs
Scanning dependencies of target gtest
Scanning dependencies of target clang_tidy_findable
Scanning dependencies of target benchmark_beluga
[  2%] Generating API documentation with Doxygen
/bin/sh: 1: DOXYGEN_EXECUTABLE-NOTFOUND: not found
make[2]: *** [docs/CMakeFiles/beluga_docs.dir/build.make:58: docs/CMakeFiles/beluga_docs] Error 127
make[1]: *** [CMakeFiles/Makefile2:1195: docs/CMakeFiles/beluga_docs.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Platform (please complete the following information):**
 - OS: Ubuntu Focal, ROS noetic
 - `Beluga` version: ed7dfe38bcab566a900b636378e7dbf05981120a

### How to reproduce
_List steps to reproduce the issue:_
1. make sure doxygen is not installed
2. clone into catkin_ws/src
3. catkin_make_isolated

_Code snippets or minimal examples are always helpful, if not necessary._

**Expected behavior**
_A clear and concise description of what you expected to happen._
No failures.

**Actual behavior**
_A clear and concise description of what actually happened._
see above

### Additional context
_Any other information you think could be meaningful to this issue._

--- a/.github/workflows/ci_pipeline.yml
+++ b/.github/workflows/ci_pipeline.yml
@@ -57,22 +57,14 @@ jobs:
       - name: Install debian packages
         run: >-
           sudo apt-get update && sudo apt-get install -y \
-            clang-tidy \
-            doxygen \
             git \
-            graphviz \
             lcov \
-            perl \
             python3-colcon-common-extensions \
             python3-colcon-coveragepy-result \
+            python3-colcon-lcov-result \
             python3-colcon-mixin \
             python3-pip \
-            python3-vcstool \
-            texlive-base \
-            texlive-binaries
-
-      - name: Install python packages
-        run: pip install colcon-lcov-result==0.5.0
+            python3-vcstool
 
       - name: Checkout repository
         uses: actions/checkout@v3
@@ -89,7 +81,7 @@ jobs:
         run: |
           . /opt/ros/${{ matrix.ros_distro }}/setup.sh
           rosdep update
-          rosdep install -r --from-paths src/ --default-yes --ignore-src --dependency-types build --dependency-types test
+          rosdep install -r --from-paths src -y -i -t build -t test -t doc
           colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mixin-repository/c75e4b34a3959524564afb584e2aa33c7eec323c/index.yaml
           colcon mixin update default
 

--- a/beluga/docs/CMakeLists.txt
+++ b/beluga/docs/CMakeLists.txt
@@ -1,22 +1,26 @@
 find_package(Doxygen)
 
-include(FetchContent)
+if(DOXYGEN_FOUND)
+  include(FetchContent)
 
-FetchContent_Declare(
-  doxygen-awesome-css
-  GIT_REPOSITORY https://github.com/jothepro/doxygen-awesome-css
-  GIT_TAG        v2.2.0
-  SOURCE_DIR     ${CMAKE_CURRENT_BINARY_DIR}/doxygen-awesome
-)
+  FetchContent_Declare(
+    doxygen-awesome-css
+    GIT_REPOSITORY https://github.com/jothepro/doxygen-awesome-css
+    GIT_TAG        v2.2.0
+    SOURCE_DIR     ${CMAKE_CURRENT_BINARY_DIR}/doxygen-awesome
+  )
 
-FetchContent_MakeAvailable(doxygen-awesome-css)
+  FetchContent_MakeAvailable(doxygen-awesome-css)
 
-add_custom_target(beluga_docs
-  ALL
-  COMMAND
-    DOXYGEN_OUTPUT_DIRECTORY=${CMAKE_CURRENT_BINARY_DIR}
-    DOXYGEN_HTML_STYLESHEET=${CMAKE_CURRENT_BINARY_DIR}/doxygen-awesome/doxygen-awesome.css
-    ${DOXYGEN_EXECUTABLE} ./Doxyfile
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-  COMMENT "Generating API documentation with Doxygen"
-  VERBATIM)
+  add_custom_target(beluga_docs
+    ALL
+    COMMAND
+      DOXYGEN_OUTPUT_DIRECTORY=${CMAKE_CURRENT_BINARY_DIR}
+      DOXYGEN_HTML_STYLESHEET=${CMAKE_CURRENT_BINARY_DIR}/doxygen-awesome/doxygen-awesome.css
+      ${DOXYGEN_EXECUTABLE} ./Doxyfile
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMENT "Generating API documentation with Doxygen"
+    VERBATIM)
+else()
+  message(WARNING "Documentation skipped, doxygen not found.")
+endif()

--- a/beluga/package.xml
+++ b/beluga/package.xml
@@ -25,6 +25,10 @@
   <test_depend>gtest</test_depend>
   <test_depend>libgmock-dev</test_depend>
 
+  <doc_depend>doxygen</doc_depend>
+  <doc_depend>graphviz</doc_depend>
+  <doc_depend>texlive-latex-base</doc_depend>
+
   <export>
     <build_type>cmake</build_type>
   </export>

--- a/docker/images/humble/Dockerfile
+++ b/docker/images/humble/Dockerfile
@@ -20,18 +20,15 @@ RUN apt-get update \
   && apt-get install --no-install-recommends -y \
     ccache \
     curl \
-    doxygen \
     gdb \
     git \
     lcov \
     linux-cloud-tools-generic \
     linux-tools-common \
     linux-tools-generic \
-    perl \
     python3-colcon-coveragepy-result \
+    python3-colcon-lcov-result \
     python3-pip \
-    texlive-base \
-    texlive-binaries \
     tmux \
   && rm -rf /var/lib/apt/lists/*
 
@@ -48,7 +45,6 @@ RUN apt-get update \
   && rm -rf /var/lib/apt/lists/*
 
 RUN pip install \
-  colcon-lcov-result==0.5.0 \
   evo==1.21.0 \
   pre-commit==2.20.0 \
   # Note: Installing timemory takes a long time.
@@ -92,7 +88,8 @@ COPY --from=cacher --chown=$USER:$GROUP /tmp/ws/ $USER_WS/
 RUN sudo apt-get update \
   && . /opt/ros/humble/setup.sh \
   && rosdep update \
-  && rosdep install -i --from-path src -y \
+  && rosdep install -i -y --from-path src \
+  && rosdep install -i -y --from-path src -t doc \
   && sudo rm -rf /var/lib/apt/lists/*
 
 COPY --from=cacher --chown=$USER:$GROUP /ws/ $USER_WS/

--- a/docker/images/noetic/Dockerfile
+++ b/docker/images/noetic/Dockerfile
@@ -2,15 +2,19 @@ FROM ros:noetic-ros-base-focal AS cacher
 
 WORKDIR /ws/src
 
-COPY docker/files/noetic.repos .
 RUN apt-get update \
-  && apt-get install --no-install-recommends -y git python3-vcstool \
-  && vcs import < noetic.repos
+  && apt-get install --no-install-recommends -y \
+    git \
+    python3-vcstool \
+  && rm -rf /var/lib/apt/lists/*
+
+COPY docker/files/noetic.repos .
+RUN vcs import < noetic.repos
 
 COPY . beluga/
 RUN touch \
-   beluga/beluga_benchmark/CATKIN_IGNORE \
-   beluga/beluga_system_tests/CATKIN_IGNORE
+  beluga/beluga_benchmark/CATKIN_IGNORE \
+  beluga/beluga_system_tests/CATKIN_IGNORE
 
 RUN mkdir -p /tmp/ws/src \
   && find ./ -name "package.xml" | xargs cp --parents -t /tmp/ws/src \
@@ -26,22 +30,17 @@ RUN apt-get update \
   && apt-get install --no-install-recommends -y \
     ccache \
     curl \
-    doxygen \
     gdb \
     git \
-    graphviz \
     lcov \
     linux-cloud-tools-generic \
     linux-tools-common \
     linux-tools-generic \
-    perl \
     python3-colcon-common-extensions \
-    python3-colcon-mixin \
     python3-colcon-coveragepy-result \
     python3-colcon-lcov-result \
+    python3-colcon-mixin \
     python3-pip \
-    texlive-base \
-    texlive-binaries \
     tmux \
   && rm -rf /var/lib/apt/lists/*
 
@@ -56,21 +55,10 @@ RUN apt-get update \
   && apt-get install --no-install-recommends -y \
     ros-rolling-ament-copyright ros-rolling-ament-uncrustify \
     ros-rolling-ament-cpplint ros-rolling-ament-flake8 \
-    ros-rolling-ament-pep257
+    ros-rolling-ament-pep257 \
+  && rm -rf /var/lib/apt/lists/*
 
 RUN sed -i '/^source.*/ s%^%source "/opt/ros/rolling/setup.bash"\n%' /ros_entrypoint.sh
-
-# Fix for the RViz black screen issue.
-# TODO(nahuel): Remove this when the Mesa version available
-# in Ubuntu 22.04 is 22.3.5 or greater.
-# See https://github.com/Ekumen-OS/beluga/issues/123
-RUN apt-get update \
-  && apt-get install --no-install-recommends -y \
-    software-properties-common \
-  && add-apt-repository -y ppa:kisak/kisak-mesa \
-  && apt-get install --no-install-recommends -y \
-    libegl-mesa0 libglx-mesa0 \
-  && rm -rf /var/lib/apt/lists/*
 
 RUN pip install \
   evo==1.21.0 \
@@ -117,6 +105,7 @@ RUN sudo apt-get update \
   && . /opt/ros/noetic/setup.sh \
   && rosdep update \
   && rosdep install -i -y --from-path src \
+  && rosdep install -i -y --from-path src -t doc \
   && sudo rm -rf /var/lib/apt/lists/*
 
 COPY --from=cacher --chown=$USER:$GROUP /ws/ $USER_WS/

--- a/docker/images/rolling/Dockerfile
+++ b/docker/images/rolling/Dockerfile
@@ -12,7 +12,7 @@ RUN mkdir -p /tmp/ws/src \
   && find ./ -name "COLCON_IGNORE" | xargs cp --parents -t /tmp/ws/src \
   || true
 
-FROM ros:rolling-ros-base-jammy as builder
+FROM ros:rolling-ros-base-jammy AS builder
 
 ENV DEBIAN_FRONTEND noninteractive
 
@@ -20,18 +20,15 @@ RUN apt-get update \
   && apt-get install --no-install-recommends -y \
     ccache \
     curl \
-    doxygen \
     gdb \
     git \
     lcov \
     linux-cloud-tools-generic \
     linux-tools-common \
     linux-tools-generic \
-    perl \
     python3-colcon-coveragepy-result \
+    python3-colcon-lcov-result \
     python3-pip \
-    texlive-base \
-    texlive-binaries \
     tmux \
   && rm -rf /var/lib/apt/lists/*
 
@@ -48,7 +45,6 @@ RUN apt-get update \
   && rm -rf /var/lib/apt/lists/*
 
 RUN pip install \
-  colcon-lcov-result==0.5.0 \
   evo==1.21.0 \
   pre-commit==2.20.0 \
   # Note: Installing timemory takes a long time.
@@ -93,7 +89,8 @@ RUN sudo apt-get update \
   && sudo apt-get -y dist-upgrade \
   && . /opt/ros/rolling/setup.sh \
   && rosdep update \
-  && rosdep install -i --from-path src -y --skip-keys 'slam_toolbox' \
+  && rosdep install -i -y --from-path src --skip-keys 'slam_toolbox' \
+  && rosdep install -i -y --from-path src -t doc \
   && sudo rm -rf /var/lib/apt/lists/*
 
 COPY --from=cacher --chown=$USER:$GROUP /ws/ $USER_WS/

